pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -419,7 +419,7 @@ if [ "$action" = edit ]; then
     exit 1
 fi
 
-SSHOPTS="$NIX_SSHOPTS -o ControlMaster=auto -o ControlPath=$tmpDir/ssh-%n -o ControlPersist=60"
+SSHOPTS=("${SSHOPTS[@]}" '-o' 'ControlMaster=auto' '-o' "ControlPath=${tmpDir}/ssh-%n" '-o' 'ControlPersist=60')
```

Comment:
This still removes the ability for the user to set `NIX_SSHOPTS`.

One solution that doesn’t anger ShellCheck is

```bash
read -ra SSHOPTS <<<"$NIX_SSHOPTS"
SSHOPTS+=(-o ControlMaster=auto -o "ControlPath=$tmpDir/ssh-%n" -o ControlPersist=60)
```

This is still not ideal: it does not provide a way for the user to pass an option containing a space. (You might hope for something like `NIX_SSHOPTS='-o foo="bar baz"'` or `NIX_SSHOPTS='-o foo=bar\ baz'` to work, but it does not.) But at least that’s not a regression from the existing script. Bash doesn’t seem to have a builtin way to split a string respecting quotes without using `eval`.

Reply:
I'd expect setting `NIX_SSHOPTS` would be unsupported, since in many cases this script ignores the original value and simply sets it to something else (for example, `NIX_SSHOPTS=$SSHOPTS runCmd nix-copy-closure --to "$buildHost" "$drv"`). If it is actually supported, I guess I could simply include it unquoted in this line and silence ShellCheck. That way, at least the code would be better, even if it's not perfect.

Alternatively, we could require `NIX_SSHOPTS` to be an array, since we already rely on Bash-only features.

Reply:
> I'd expect setting `NIX_SSHOPTS` would be unsupported, since in many cases this script ignores the original value and simply sets it to something else (for example, `NIX_SSHOPTS=$SSHOPTS runCmd nix-copy-closure --to "$buildHost" "$drv"`).

That happens after it’s already set `SSHOPTS` based on the user’s `NIX_SSHOPTS`.

> Alternatively, we could require `NIX_SSHOPTS` to be an array, since we already rely on Bash-only features.

An array can’t be stored in the environment. If you try to `export` an array from Bash, the child process sees a string.

That brings up another problem: when this script does set `NIX_SSHOPTS`, it does so for the benefit of `nix-copy-closure`, which splits it by spaces and is not compatible with Bash’s stringification of arrays.